src/hasAtLeast.ts

```diff
@@ -1,14 +1,41 @@
-import type { ReadonlyTuple } from "./_types";
+import type { IterableContainer } from "./_types";
 import { purry } from "./purry";
 
-type ArrayMinN<T, N extends number> = number extends N
-  ? // We can only compute the type for a literal number!
-    Array<T>
-  : // I don't know why we need to special-case the 0 case, but otherwise
-    // typescript complains we have a deep recursion.
-    N extends 0
-    ? Array<T>
-    : [...ReadonlyTuple<T, N>, ...Array<T>];
+type ArraySetRequired<
```

Comment:
**question:** why the rename? i think i understood `ArrayMinN` better than this

Reply:
ArrayMinN took the **item's type** as T, just like Array does (Array<T> vs ArrayMinN<T, N>), but now the function takes the **input array** as T. It works more similarly to type-fests [SetRequired](https://github.com/sindresorhus/type-fest/blob/main/source/set-required.d.ts) where T is the object that is being transformed, and the type name describes the transformation (and not the output).